hortonworks/cloudbreak
# CB 19575 determine free space on a database backup node

*Motivation:*
 We want to prevent backups which will fail. One of the reasons backup can fail is a lack of temporary storage space on a database backup node. We want to send information about free available space on a node - we know which node performs database backup - to Thunderhead where it will be checked against required storage.

*Implementation*
Available space is collected by a Salt script using Linux _df_ command in POSIX compatibility mode (so higher chances output is uniform at different installations).
We reuse "determine datalake_data_sizes" call to feed this information into Thunderhead Datalake DR since it belongs to that process.

*Test approach:*
- Start database backup
```shell
cdp datalake backup-datalake --datalake-name sartyukh-045-dl-db-backup --debug`
```
"backupId": "34f563d0-a975-424c-b02c-48a99ae28157",

- call `cb/api/v4/1/stacks/$DATALAKE$/determine_datalake_data_sizes`
```shell
https --json --headers POST http://localhost:9091/cb/api/v4/1/stacks/sartyukh-045-dl-db-backup/determine_datalake_data_sizes operationId==$OPERATION_ID x-cdp-actor-crn:$ACTOR_ID x-cdp-request-id:`uuidgen`
```
 HTTP/1.1 204 

- monitor progress in the _CBDB_ database, table _stackstatus_, row _DETERMINE_DATALAKE_DATA_SIZES_FINISHED_
- verify thunderhead database _datalakedr_ table _statusentity_internalstaterecords_ for data
```shell
datalakdr=# select substatetype, size_in_bytes from datalakedrstatusentity_internalstaterecords where datalakedrstatusentity_droperationid = '34f563d0-a975-424c-b02c-48a99ae28157';
```
|  DATABASE_BACKUP_STORAGE_VALIDATION |  81828952 |
